pkg/api/store/cluster/cluster_store.go

```diff
@@ -22,6 +26,101 @@ type Store struct {
 	mu           sync.Mutex
 }
 
+func SetClusterStore(ctx context.Context, schema *types.Schema, mgmt *config.ScaledContext) {
+	t := &transform.Store{
+		Store: schema.Store,
+		Transformer: func(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}, opt *types.QueryOptions) (map[string]interface{}, error) {
+			data, err := transformSetNilSnapshotFalse(data)
+			if err != nil {
+				return nil, err
+			}
+
+			return data, nil
+		},
+	}
+
+	s := &Store{
+		Store: t,
+	}
+
+	schema.Store = s
+}
+
+func transformSetNilSnapshotFalse(data map[string]interface{}) (map[string]interface{}, error) {
+	appliedVal, ok := data["appliedSpec"]
+	if ok {
+		appliedAsMap, ok := appliedVal.(map[string]interface{})
+		if !ok {
+			return nil, errors.New("Error parsing applied spec")
+		}
+
+		appliedData, err := transformSetNilSnapshotFalseForTree(appliedAsMap)
+		if err != nil {
+			return nil, err
+		}
+		data["appliedSpec"] = appliedData
+	}
+
+	return transformSetNilSnapshotFalseForTree(data)
+}
+
+func transformSetNilSnapshotFalseForTree(data map[string]interface{}) (map[string]interface{}, error) {
+	configVal, ok := data["rancherKubernetesEngineConfig"]
+	if !ok {
+		return data, nil
+	}
+
+	configAsMap, ok := configVal.(map[string]interface{})
```

Comment:
for getting the nested value, `getValueN` can be used. Example: https://github.com/rancher/types/blob/master/mapper/scheduling.go#L200

Reply:
And for setting the nested value, `convert.PutValue` can be used: https://github.com/rancher/norman/blob/master/store/proxy/proxy_store.go#L305